contracts/KeysManager.sol

```diff
@@ -360,6 +367,9 @@ contract KeysManager is EternalStorage, IKeysManager {
         if (!IPoaNetworkConsensus(poaNetworkConsensus()).addValidator(_key, true)) {
             return false;
         }
+        if (hasMiningKeyBeenRemoved(_key)) {
+            return false;
+        }
```

Comment:
These lines should be placed before calling of `IPoaNetworkConsensus.addValidator`. Let's move them three lines higher.